README.md

```diff
@@ -1,5 +1,103 @@
-# ecs
+# ECS
 
 [![Build Status](https://travis-ci.org/EngoEngine/ecs.svg?branch=master)](https://travis-ci.org/EngoEngine/ecs)
 
-This is our implementation of the "Entity Component System" model in Go. It was designed to be used in `engo`, however it is not dependent on any other packages so is able to be used wherever!
+## What's ECS?
+ECS stands for Entity-Component-System paradigm. More information can be found [here](https://en.wikipedia.org/wiki/Entity_component_system). 
+
+Basically, you have a `World`, which consists of a series of different `System`s and `Entity`s. The `Entity`s are 
+modular, and can store values within `Component`s. All the logic happens within `System`s: so whenever you need to
+add / alter / update an `Entity`, you will usually do so from within a `System`. If you need to pass information
+from one `System` to another, you can store intermediary values within a `Component` within that `Entity`, and read
+that value back from within the other `System`. 
+
+## So how does it work?
+This currently forms the core of the game-engine [engo](https://github.com/EngoEngine/engo). 
+
+If you want to use it without using `engo`, you can continue reading. 
+
+
+### Globally 
+
+```go
+// Defining it, and calling the required initialization function
+world := ecs.World{}
+world.New()
+
+// Then you add a series of Systems - you can do this at any point
+world.AddSystem(mySystem1)
+world.AddSystem(mySystem2)
+
+// After that, you can add a few Entities to the World. The first is being added to mySystem1, 
```

Comment:
If we are going to use the plural `Entities` here, we should make it lowercase "E"